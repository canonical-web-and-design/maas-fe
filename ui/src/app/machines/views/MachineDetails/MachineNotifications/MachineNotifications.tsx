import type { ReactNode } from "react";

import { Notification } from "@canonical/react-components";
import type { NotificationProps } from "@canonical/react-components";

type MachineNotification = {
  active: boolean;
  content: ReactNode;
  severity?: NotificationProps["severity"];
  title?: string;
};

type Props = {
  notifications: MachineNotification[];
};

const MachineNotifications = ({ notifications }: Props): JSX.Element => {
  const notificationList = notifications.reduce<ReactNode[]>(
    (collection, { active, content, severity, title }, i) => {
      if (active) {
        collection.push(
          <Notification key={i} severity={severity} title={title}>
            {content}
          </Notification>
        );
      }
      return collection;
    },
    []
  );

  return (
    <section className="p-strip u-no-padding--top u-no-padding--bottom">
      <div className="row">{notificationList}</div>
    </section>
  );
};

export default MachineNotifications;
