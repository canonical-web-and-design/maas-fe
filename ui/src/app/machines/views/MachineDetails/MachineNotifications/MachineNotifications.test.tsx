import { mount } from "enzyme";

import MachineNotifications from "./MachineNotifications";

describe("MachineNotifications", () => {
  it("can render", () => {
    const wrapper = mount(
      <MachineNotifications
        notifications={[
          {
            active: true,
            content:
              "Editing is currently disabled because no rack controller is currently connected to the region.",
            severity: "negative",
            title: "Error:",
          },
        ]}
      />
    );
    expect(wrapper.find("MachineNotifications")).toMatchSnapshot();
  });

  it("ignores inactive notifications", () => {
    const wrapper = mount(
      <MachineNotifications
        notifications={[
          {
            active: false,
            content: "Don't show me!",
            severity: "negative",
            title: "Error:",
          },
        ]}
      />
    );
    expect(wrapper.find("Notification").exists()).toBe(false);
  });
});
