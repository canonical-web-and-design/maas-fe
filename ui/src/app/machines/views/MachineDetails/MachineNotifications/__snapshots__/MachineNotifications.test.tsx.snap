// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineNotifications can render 1`] = `
<MachineNotifications
  notifications={
    Array [
      Object {
        "active": true,
        "content": "Editing is currently disabled because no rack controller is currently connected to the region.",
        "severity": "negative",
        "title": "Error:",
      },
    ]
  }
>
  <section
    className="p-strip u-no-padding--top u-no-padding--bottom"
  >
    <div
      className="row"
    >
      <Notification
        key="0"
        severity="negative"
        title="Error:"
      >
        <div
          className="p-notification--negative"
        >
          <div
            className="p-notification__content"
          >
            <h5
              className="p-notification__title"
              data-test="notification-title"
            >
              Error:
            </h5>
            <p
              className="p-notification__message"
            >
              Editing is currently disabled because no rack controller is currently connected to the region.
            </p>
          </div>
        </div>
      </Notification>
    </div>
  </section>
</MachineNotifications>
`;
